src/main/java/it/polimi/telco/model/Order.java

```diff
@@ -0,0 +1,116 @@
+package it.polimi.telco.model;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.Date;
+import java.util.List;
+
+@Entity
+@Table(name = "order", schema = "telco")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull
+    private Date dateOfCreation;
+
+    @NotNull
+    @OneToOne
+    private User user;
+
+    @NotNull
+    @OneToOne
+    private ServicePackage servicePackage;
+
+    @NotNull
+    @OneToOne
+    private ValidityPeriod validityPeriod;
+
+    @NotNull
+    @OneToMany
+    List<Product> optionalProductList;
+
+    @NotNull
+    public double totalPrice;
+
+    @NotNull
+    public Date subscriptionStartDate;
```

Comment:
We could even link order to a subscription. In that case we wouldn't need user, servicePackage, validityPeriod, optionalProductList and subscriptionStartDate

Reply:
I taught to do like this first but to be honest, I think this is a more clean way of doing it( also from what I read in the documentation, to me it sounded like each of these subscription attributes has to be also included in the order and populate the order from the subscription). Did you understand differently?